context-free.md

```diff
@@ -1,22 +1,25 @@
 # Context Free
 
 ##  Who is this programming language for?
-
+People who want to create digital art. 
 
 ## What is easy to do in this language? Why is it easy?
-
+It is easy to create images recursively with the looping syntax - things like spiral, snowflakes and trees. It's also easy to implement randomness by creating different versions of a shape and assigning weights to them. 
 
 ## What is hard to do in this language? Why is it hard?
-
+It is hard to draw precise images without patterns in this language. Getting shapes to appear in the right position is like playing around with CSS - it involves quite a bit of experimentation - and only having primitive shapes to draw things is quite limiting. Images that have many unique parts to them would be very hard and tedious to draw. For instance, a portrait of a face (even if we assume it's perfectly symmetric) would require hundreds of unique lines and colors. 
 
 ## How did you learn how to program in this language?
+At first, I read through the documentation to get a feel for the syntax and "words" of the language. Once I had understood the basics, I moved onto playing around with the examples, experimenting with different values to see the results and understand how the language works. Whenever I was unsure of how something worked I consulted the documentation on Github. 
 
 
 ## What is the underlying _computational model_ for this programming language? 
 _We don't yet have a great definition of the term "computational model". 
 For now, try to come up with the clearest, most concise explanation of what 
 happens when a ContextFree program runs._
 
+The program renders the shapes in the order that the syntax references them, from the top to the bottom. In the more complicated programs we can see this process unfold. 
```

Comment:
Maybe adding something about the use of "rules" and how they can be used recursively.